### PR TITLE
Model Base Update

### PR DESCRIPTION
**Pull request by devotox**
Document the different attribute options
- Ability to transform values
- Ability to have a custom validation
- Ability to have a default value if no value passed in

### CODE REVIEW

File: lib/model_utilities.js [hunk 1]
```diff
@@ -22,6 +21,15 @@ module.exports.isObject = (value) => {
 };
 
 /**
+ * @param  { any } value [ value to test is an array ]
+ * @return { boolean } [ true / false if parameter is a array ]
```
Comment (rafaelalvessa) [inline lib/model_utilities.js]: is a array => is a_n_ array

File: tests/AbstractModel.js [hunk 2]
```diff
@@ -258,6 +257,136 @@ describe('AbstractModel', () => {
...
+			expectation.test_attributes.test_key.custom = [
+				(value) => {
+					return value === 'test' ? null : 'value should equal test';
+				}
+			];
+
+			let TestModel = new Model(expectation.test_attributes, expectation.test_values);
+
+			let valid = TestModel.isValid();
+			chai.assert(valid === true, 'Incorrect Validation');
```
Comment (rafaelalvessa) [inline tests/AbstractModel.js:312]: Why `'Incorrect Validation'` if `valid` is `true`?

File: tests/AbstractModel.js [hunk 3]
```diff
@@ -258,6 +257,136 @@ describe('AbstractModel', () => {
...
+					return !!value;
+				}
+			};
+
+			expectation.test_values.test_key2 = '0.2';
+
+			let TestModel = new Model(expectation.test_attributes, expectation.test_values);
+
+			let valid = TestModel.isValid();
+			chai.assert(valid === true, 'Incorrect Validation');
```
Comment (rafaelalvessa) [inline tests/AbstractModel.js:356]: Same thing here.

File: tests/AbstractModel.js (hunk 2 above)
Comment (devotox) [reply tests/AbstractModel.js:312]: becuase if it fails then that is the Incorrect validation

File: tests/AbstractModel.js (hunk 3 above)
Comment (devotox) [reply tests/AbstractModel.js:356]: becuase if it fails then that is the Incorrect validation

**Comment by bill-edwards** [comment]
Looks good.

**Review by rafaelalvessa — APPROVED** [review]

**Review by rafaelalvessa — APPROVED** [review]

**Pull request merged**